request_test.go

```diff
@@ -22,7 +22,7 @@ func TestClient_RawRequest(t *testing.T) {
 	}
 	c := &Client{}
 	for _, h := range validAPIHosts {
-		c.url = &url.URL{Path: h}
+		c.url, _ = url.Parse(h)
```

Comment:
I still don't like that we are throwing away the error here. I think we should ensure the URL did, in fact, parse, and Fatalf if it did not.